src/api/controller/api/enrichment.js

```diff
@@ -57,6 +59,13 @@ export const putEnrichment = async (ctx, id) => {
 export const deleteEnrichment = async (ctx, id) => {
     try {
         const enrichment = await ctx.enrichment.findOneById(id);
+        const activeJob = await getActiveJob();
+        if (
+            activeJob?.data?.jobType === ENRICHER &&
+            activeJob?.data?.id === id
+        ) {
+            cancelJob(ctx, ENRICHER);
```

Comment:
This is called asynchronously and modified `ctx.status`, possibly after we already answered. I have the feeling this could break. Can we have a cancelJob function that does just that and let the caller handle return codes?

Reply:
Yes, you're right, well spotted